scapy/arch/linux.py

```diff
@@ -224,16 +224,22 @@ def read_routes():
         return []
     routes = []
     s=socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
-    ifreq = ioctl(s, SIOCGIFADDR,struct.pack("16s16x", LOOPBACK_NAME.encode("utf8")))
-    addrfamily = struct.unpack("h",ifreq[16:18])[0]
-    if addrfamily == socket.AF_INET:
-        ifreq2 = ioctl(s, SIOCGIFNETMASK,struct.pack("16s16x", LOOPBACK_NAME.encode("utf8")))
-        msk = socket.ntohl(struct.unpack("I",ifreq2[20:24])[0])
-        dst = socket.ntohl(struct.unpack("I",ifreq[20:24])[0]) & msk
-        ifaddr = scapy.utils.inet_ntoa(ifreq[20:24])
-        routes.append((dst, msk, "0.0.0.0", LOOPBACK_NAME, ifaddr, 1))
-    else:
-        warning("Interface lo: unkown address family (%i)"% addrfamily)
+    try:
+        ifreq = ioctl(s, SIOCGIFADDR, struct.pack("16s16x", scapy.consts.LOOPBACK_NAME.encode("utf8")))
+        addrfamily = struct.unpack("h", ifreq[16:18])[0]
+        if addrfamily == socket.AF_INET:
+            ifreq2 = ioctl(s, SIOCGIFNETMASK, struct.pack("16s16x", scapy.consts.LOOPBACK_NAME.encode("utf8")))
+            msk = socket.ntohl(struct.unpack("I", ifreq2[20:24])[0])
+            dst = socket.ntohl(struct.unpack("I", ifreq[20:24])[0]) & msk
+            ifaddr = scapy.utils.inet_ntoa(ifreq[20:24])
+            routes.append((dst, msk, "0.0.0.0", scapy.consts.LOOPBACK_NAME, ifaddr, 1))
+        else:
+            warning("Interface lo: unknown address family (%i)" % addrfamily)
+    except IOError as err:
```

Comment:
Looks way better now thanks! I have the following small request. Except that, it looks perfect !

Could you use `scapy.consts.LOOPBACK_NAME` instead of `lo` at (your) lines 237, 240 and 242 ? That will be consistent with the code.

Reply:
uhm... yes. I already changed it and whipped it during the commit cleanup... (well - this 'single purpose' has a strong karma)